### PR TITLE
Integration specs for views

### PR DESCRIPTION
**Pull request by Mofuhidy**
# What we did in this PR:

1.  We solved the n+1 problem

> before
Screenshot 2023-10-03 at 4 08 19 PM

> after
Screenshot 2023-10-03 at 4 08 25 PM

2. Add testing for the views.

### CODE REVIEW

File: app/views/posts/_comment_card.html.erb [hunk 1]
```diff
@@ -1,7 +1,8 @@
 <div class="comment-container">
 
     <% if comments.any? %>
-        <% comments.each do |comment| %>
+        <% comments.includes(:user).each do |comment| %>
+
```
Comment (ShadyShawkat) [inline app/views/posts/_comment_card.html.erb]: - [x] Kindly notice that fixing the N+1 issue in the views doesn't actually fix it due to the fact that data had already been fetched from the database. To fix the problem, you'll need to implement the solution inside of the controller which then takes effect when the data is fetched which fixes the problem. Kindly use the `includes` method inside of the Posts controller and not in the views.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @Mofuhidy, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/posts/_comment_card.html.erb (hunk 1 above)
Comment (Mofuhidy) [reply app/views/posts/_comment_card.html.erb]: Ok Thanks

**Review by AbelG101 — APPROVED** [review]
# Hi @Mofuhidy & @aradradev 👋, 

**Your project is complete!  There is nothing else to say other than... it's time to merge it. 
Congratulations! 🎉 🥳🥇**

## Highlights
- Associated comments for all posts are eagerly loaded ✔
- Correct use of Git Flow ✔
- All linter checks are passing ✔
- Descriptive PR title with a great summary ✔

## Optional suggestions
- N/A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**